src/write.rs

```diff
@@ -450,6 +465,66 @@ mod tests {
         assert_eq!(w.oplog.len(), 2);
     }
 
+    #[test]
+    fn wait_test() {
+        use std::sync::{Arc, Barrier};
+        use std::thread;
+        let (mut w, _r) = crate::new::<i32, _>();
+
+        // Case 1: If epoch is set to default.
+        let test_epochs: crate::Epochs = Default::default();
+        let mut test_epochs = test_epochs.lock().unwrap();
+        // since there is no epoch to waiting for, wait function will return immediately.
+        w.wait(&mut test_epochs);
+
+        // Case 2: If one of the reader is still reading(epoch is odd and count is same as in last_epoch)
+        // and wait has been called.
+        let held_epoch = Arc::new(AtomicUsize::new(1));
+
+        w.last_epochs = vec![2, 2, 1];
+        let mut epochs_slab = Slab::new();
+        epochs_slab.insert(Arc::new(AtomicUsize::new(2)));
+        epochs_slab.insert(Arc::new(AtomicUsize::new(2)));
+        epochs_slab.insert(Arc::clone(&held_epoch));
+
+        let barrier = Arc::new(Barrier::new(2));
+
+        let is_waiting = Arc::clone(&w.is_waiting);
```

Comment:
You can probably just assert that this is `false` here rather than doing it in both threads before the barrier wait below. Arguably you don't even need that pre-assert.